src/rgw/rgw_http_client.h

```diff
@@ -102,6 +102,52 @@ class RGWHTTPClient
   int get_req_retcode();
 };
 
+
+class RGWHTTPHeadersCollector : public RGWHTTPClient {
+public:
+  /* Case insensitive comparator for containers carrying HTTP headers. */
+  struct CILess : public std::binary_function<std::string, std::string, bool> {
+    bool operator()(const std::string& lhs,
+                    const std::string& rhs) const {
+      return ::strcasecmp(lhs.c_str(), rhs.c_str()) < 0 ;
```

Comment:
This is has been changed in https://github.com/ceph/ceph/pull/8657/commits/45494d40a7b9c648e817b0e00189e07363a3d6c9 to reuse `ltstr_nocase` of `rgw_string.h`. This comparator is quite popular across the code.